### ISSUE TITLE
Doesn't work in Next.js 14 (production mode)

### ISSUE THREAD

**Issue by humanua** [akiran/react-slick#2336]
In version 0.30.1 in development mode in Next.js 14 it works well, but when I build my application (production mode) the slider doesn't work. but everything works fine in version 0.29.0

**Comment by Davron129** [comment]
Same issue, I got this error after build command:

`Type error: 'Slider' cannot be used as a JSX component.
  Its type 'typeof Slider' is not a valid JSX element type.`

**Comment by 20manas** [comment]
I was having the same issue. In my case, what the fixed the issue was removing `@types/react-slick`

**Comment by laolee010126** [comment]
Do not forget add 'use client'

**Comment by refex** [comment]
same here, next14

**Comment by daibergm** [comment]
Same issue, I'm using next V14.1.4

**Comment by GeorgeArubi** [comment]
> I was having the same issue. In my case, what the fixed the issue was removing `@types/react-slick`

So how did you use the Slider without type definitions? I'm assuming you're using typescript right?

**Comment by 20manas** [comment]
> > I was having the same issue. In my case, what the fixed the issue was removing `@types/react-slick`
> 
> So how did you use the Slider without type definitions? I'm assuming you're using typescript right?

Yes I am using typescript. The types aren't available, but what is accepted by the components is in the docs, so I used that.

**Comment by linucks** [comment]
Same issue here. Removing the type declartion doesn't help ("react-slick": "^0.30.2", "next": "^14.2.3")

**Comment by songxiaoliang** [comment]
Same issue here. Removing the type declartion doesn't help ("react-slick": "^0.30.2", "next": "^14.2.3")

**Comment by aliffazfar** [comment]
You can fix by using this specific version in your package.json:
"@types/react": "18.2.21",
"@types/react-dom": "18.2.7"

**Comment by ReactDev81** [comment]
I am using the following versions: "next": "14.2.6", "react": "^18", "react-slick": "^0.29.0", and "slick-carousel": "^1.8.1". I am encountering an issue.

![image](https://github.com/user-attachments/assets/b8082acc-1432-4d1e-b0af-a6a2506cb474)

So just using 'use client' at the top of the component and the issue is resolved

**Comment by EmmanuelVictor62** [comment]
> Same issue, I got this error after build command:
> 
> `Type error: 'Slider' cannot be used as a JSX component. Its type 'typeof Slider' is not a valid JSX element type.`

@Davron129, were you able to solve this issue?

**Comment by kferolinomersoft** [comment]
I tried to use npm instead of yarn and the error disappearred.

**Comment by a-fatah** [comment]
I am also facing the same issue in a react app. 
```
'Slider' cannot be used as a JSX component.
      Its type 'typeof Slider' is not a valid JSX element type.
        Types of construct signatures are incompatible.
          Type 'new (props: Settings) => Slider' is not assignable to type 'new (props: any, deprecatedLegacyContext?: any) => Component<any, any, any>'.
            Property 'refs' is missing in type 'Slider' but required in type 'Component<any, any, any>'.
```

**Comment by Opeyemi-Omotayo** [comment]
```
'Slider' cannot be used as a JSX component.
  Its type 'typeof Slider' is not a valid JSX element type.
    Types of construct signatures are incompatible.
      Type 'new (props: Settings) => Slider' is not assignable to type 'new (props: any, deprecatedLegacyContext?: any) => Component<any, any, any>'.
        Property 'refs' is missing in type 'Slider' but required in type 'Component<any, any, any>'.ts(2786)
index.d.ts(1040, 9): 'refs' is declared here.

```

I am using next 14.1.4 and react-slick 0.30.2 and this solved the issue for me

`declare module 'react-slick' {
    import { Component } from 'react';
  
    export interface ResponsiveSetting {
      breakpoint: number;
      settings: Partial<Settings>;
    }
  
    export interface Settings {
      infinite?: boolean;
      speed?: number;
      slidesToShow?: number;
      slidesToScroll?: number;
      initialSlide?: number;
      arrows?: boolean;
      prevArrow?: React.ReactNode;
      nextArrow?: React.ReactNode;
      responsive?: ResponsiveSetting[];
      afterChange?: (currentSlide?: number) => void;
      asNavFor?: any;
      [key: string]: any;
    }
  
    export default class Slider extends Component<Settings> {
      slickNext(): void;
      slickPrev(): void; 
      slickGoTo(slide: number, dontAnimate?: boolean): void; 
    }
  }
  `
  i just created a `declaration.d.ts` file

**Comment by yonroku46** [comment]
i  just add for **package.json** file
`"resolutions": {
    "@types/react": "^18.2.43"
  }`

**Comment by shehan-mark** [comment]
I also faced this issue, but not with Next.js. My case was just a latest react project created using Vite. Underlying problem is like most of here have mentioned, is related to `react-slick` is trying to depend on the latest `@types/react` version (which is 19 in my case).  `react-slick` should mention a specific peer dependency version instead of trying fetch/ match latest of `@types/react` versions.

How I solved this issue is by uninstalling `@types/react-slick` from my project and creating a standalone type declaration file for `react-slick` package. 

[This](https://github.com/akiran/react-slick/issues/2336#issuecomment-2589213857) answer helped me to solve this issue on my side.

**Comment by will-hwd** [comment]
Creating the `declaration.d.ts` file fixed my issue, **but** i'm using vite instead of next and had to create the file in `project/src/react-slick.d.ts`
